lib/Runtime/PlatformAgnostic/IPlatformAgnosticResource.h

```diff
@@ -0,0 +1,63 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#ifndef RUNTIME_PLATFORM_AGNOSTIC_COMMON_IFINALIZABLE
+#define RUNTIME_PLATFORM_AGNOSTIC_COMMON_IFINALIZABLE
+
+namespace PlatformAgnostic
+{
+namespace Resource
+{
+    // Interface
+    class IPlatformAgnosticResource
+    {
+    public:
+        virtual void Cleanup() = 0;
+    };
+
+    template <typename T>
+    class AutoPtr
+    {
+    private:
+        const T *pointer;
+    public:
+        AutoPtr(const T *ptr) : pointer(ptr) {}
+        ~AutoPtr()
+        {
+            if (pointer != nullptr)
+            {
+                delete pointer;
+                pointer = nullptr;
+            }
+        }
+
+        void setPointer(const T *ptr)
+        {
+            pointer = ptr;
+        }
+    };
+
+    template <typename T>
+    class StringBufferAutoPtr
+    {
+    private:
+        const T *strBuf;
+    public:
+        StringBufferAutoPtr(const T *buffer) : strBuf(buffer) {}
+        ~StringBufferAutoPtr()
+        {
+            delete[] strBuf;
```

Comment:
I'm updating to check for nullptr first. Otherwise leaving alone.

---
In reply to: [141508964](https://github.com/Microsoft/ChakraCore/pull/3809#discussion_r141508964) [](ancestors = 141508964)